keychain/src/utils/account.ts

```diff
@@ -117,6 +117,8 @@ class Account extends BaseAccount {
           Storage.update(this.selector, {
             status: Status.COUNTERFACTUAL,
           });
+          // async deployed, poll for contract
+          setTimeout(() => this.sync(), 5000);
```

Comment:
Hmm I'm not sure I understand the specific flow where this is necessary? Is it if the deploy comes from another application?